server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -3059,4 +3060,20 @@ public ShardLongFieldRange getRawFieldRange(String field) {
         return ShardLongFieldRange.UNKNOWN;
     }
 
+    @Override
+    public long getCurrentGeneration() {
+        final ReferenceManager<ElasticsearchDirectoryReader> referenceManager = getReferenceManager(SearcherScope.EXTERNAL);
+        ElasticsearchDirectoryReader current = null;
+        try {
+            current = referenceManager.acquire();
+            return current.getIndexCommit().getGeneration();
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        } finally {
+            if (current != null) {
+                ElasticsearchDirectoryReader toClose = current;
+                IOUtils.closeWhileHandlingException(() -> referenceManager.release(toClose));
+            }
+        }
+    }
```

Comment:
What is the difference between this and simply returning `store.readLastCommittedSegmentsInfo().getLastGeneration()`?

Reply:
The code has changed in this area.